### PR TITLE
Refactored FAQ content to make all answer preview fields optional

### PR DESCRIPTION
**Pull request by sraby**
This PR simplifies our NoRent FAQ content through refactoring the array we store it in. The main difference here is that we now update the `Faq` type to allow for some fields to be optional.

Old type definition:
```ts
export type Faq = {
  question: string;
  category: FaqCategory;
  priorityInPreview: number;
  answerPreview: React.ReactNode;
  answerFull: React.ReactNode;
};
```

New type definition:
```ts
export type Faq = {
  question: string;
  category: FaqCategory;
  answerFull: React.ReactNode;
  previewOptions?: {
    priorityInPreview: number;
    answerPreview: React.ReactNode;
  };
};
```

The consequences of this change are that:
1. Only FAQ entries that we want to include in a preview section (like on the homepage or "The Letter" page) will need to have a numeric priority and `answerPreview` content defined. 
2. The `priorityInPreview` field will determine the order they show up on the preview section. 
3. For the full list of FAQ entries on the FAQs page, order will be determined by the order entries appear in the FAQ content array, once they are sorted into categories. Therefore, adding in a new FAQ entry to this page simply means adding a new element within the array— no hassle with defining a "priority" and shifting other entries' priorities to line up.

My hope is that these changes will set us up to have a smoother time with internationalization.

### CODE REVIEW

File: frontend/lib/norent/faqs.tsx [hunk 1]
```diff
@@ -55,7 +51,16 @@ export const ChevronIcon = () => (
...
+  const FaqsPreviewContent = FaqsContent.filter(
+    (faq) => faq.previewOptions
+  ).sort((faq1, faq2) =>
+    faq1.previewOptions?.priorityInPreview &&
+    faq2.previewOptions?.priorityInPreview
+      ? faq1.previewOptions.priorityInPreview -
+        faq2.previewOptions.priorityInPreview
+      : 0
+  );
+
```
Comment (sraby) [inline frontend/lib/norent/faqs.tsx:63]: Agh so this definitely got a little messy here... it seemed that even after applying my `filter()` to grab faq entries that had their preview options configured, I'd still get a ts error staying that `previewOptions` could possibly be undefined. Any hints as to how I could avoid that?

File: frontend/lib/norent/data/faqs-content.tsx [hunk 2]
```diff
@@ -11,9 +11,11 @@ export type FaqCategory =
 export type Faq = {
   question: string;
   category: FaqCategory;
-  priority: number;
-  answerPreview: React.ReactNode;
   answerFull: React.ReactNode;
+  previewOptions?: {
```
Comment (sraby) [inline frontend/lib/norent/data/faqs-content.tsx:15]: I feel mixed about calling this `previewOptions` but couldn't come up with an alternative. Do you have any preferences?

File: frontend/lib/norent/faqs.tsx (hunk 1 above)
Comment (toolness) [reply frontend/lib/norent/faqs.tsx:63]: ahh yeah that is annoying that typescript's type inference doesn't totally work here.  I'm not sure of an easy way to work around it unfortunately, so maybe just add a comment mentioning what you wrote here, and we can figure out an easier way to write it out later?

File: frontend/lib/norent/data/faqs-content.tsx (hunk 2 above)
Comment (toolness) [reply frontend/lib/norent/data/faqs-content.tsx:15]: hmm not really, i think it looks fine as-is though!

File: frontend/lib/norent/faqs.tsx (hunk 1 above)
Comment (sraby) [reply frontend/lib/norent/faqs.tsx:63]: Sounds good to me! Thank you

**Pull request merged**